### DISCUSSION TITLE
Foreign keys validation inside models

### DISCUSSION THREAD

**Discussion by pinkynrg** [adonisjs/core#3943] [Help] [answered]
I have the following tables: 

- users
- customers
- members
- workspaces
- seats 

following, the description of the relationships: 
- users may have more customers and vice versa through the table members
- a customer may have more workspaces 
- a workspace may have more members

The problem with this setup is the following:
I need to validate at the application level that a **workspace** that belongs to a **customer** "X" can only be associated with a **member** that belongs to a **customer** "X".

In Django, I would use model validation: https://docs.djangoproject.com/en/4.1/ref/models/instances/#django.db.models.Model.clean

Is it possible to have the same type of validator at the model level? 

Maybe do the foreign keys check in the beforeCreate hook and, if something goes wrong, throw an Exception inside the hook?

**Comment by thetutlage** [comment, marked as answer]
Yeah, you can use `beforeCreate` or `beforeSave` hooks for the same
